hw/ip/csrng/data/csrng_sec_cm_testplan.hjson

```diff
@@ -24,130 +24,247 @@
 {
   testpoints: [
     {
+      // TODO: Unlike the REGWEN register, the CTRL and ERR_CODE_TEST registers are excluded from some automated CSRs tests (CsrExclWrite).
+      // We need to ensure write-read checks are happening or alternatively test point 2) explicitly, e.g., with a directed test.
       name: sec_cm_config_regwen
-      desc: "Verify the countermeasure(s) CONFIG.REGWEN."
+      desc: '''
+            Verify the countermeasure(s) CONFIG.REGWEN.
+            Verify that:
+            1) REGWEN cannot be set back to 1 after being set to 0 once.
+            2) If REGWEN is not set, the CTRL and ERR_CODE_TEST registers cannot be modified.
+            '''
       stage: V2S
-      tests: []
+      tests: ["csrng_csr_rw"]
     }
     {
+      // TODO: Currently, we don't verify that the alert actually fires, just that alert shows in the RECOV_ALERT_STS register.
       name: sec_cm_config_mubi
-      desc: "Verify the countermeasure(s) CONFIG.MUBI."
+      desc: '''
+            Verify the countermeasure(s) CONFIG.MUBI.
+            Verify that upon writing invalid MUBI values to the CTRL register:
+            1) the DUT signals a recoverable alert and sets the correct bit in the RECOV_ALERT_STS register, and
+            2) the DUT can be configured back to a safe configuration and the RECOV_ALERT_STS register can be cleared.
+            '''
       stage: V2S
-      tests: []
+      tests: ["csrng_alert"]
     }
     {
+      // TODO: This is currently untested.
+      // The environment has support for randomly driving MuBi8False to otp_en_csrng_sw_app_read but always MuBi8True is driven.
+      // In addition, the environment needs to be extended to drive also non-valid encodings (see entropy_src).
       name: sec_cm_intersig_mubi
-      desc: "Verify the countermeasure(s) INTERSIG.MUBI."
+      desc: '''
+            Verify the countermeasure(s) INTERSIG.MUBI.
+            Verify that unless the otp_en_csrng_sw_app_read input signal is equal to MuBi8True and CTRL.SW_APP_ENABLE or CTRL.READ_INT_STATE is set to kMultiBitBool4True the DUT doesn't allow reading the genbits or the internal state from the GENBITS or INT_STATE_VAL register, respectively.
+            '''
       stage: V2S
       tests: []
     }
     {
       name: sec_cm_main_sm_fsm_sparse
-      desc: "Verify the countermeasure(s) MAIN_SM.FSM.SPARSE."
+      desc: '''
+            Verify the countermeasure(s) MAIN_SM.FSM.SPARSE.
+            The csrng_intr and csrng_err tests verify that if the FSM state is forced to an illegal state encoding 1) this is reported with a cs_fatal_err interrupt in the INTR_STATE register and 2) the corresponding bit in the ERR_CODE register is set.
+            They currently don't check whether the DUT actually triggers a fatal alert.
```

Comment:
Should this be a TODO comment instead?

Reply:
Thanks for your review @marnovandermaas . Ideally, we would add a TODO for this but it's not absolutely required, as the FPV tests check the alert connection and triggering (see next line). Because time is very pressing, I would not add further work.